src/hotspot/cpu/s390/s390.ad

```diff
@@ -837,12 +837,18 @@ void MachPrologNode::format(PhaseRegAlloc *ra_, outputStream *st) const {
   }
   st->print_cr("push_frame %d", (int)-framesize);
   st->print("\t");
+
+  // TODO: Check this!
+  // if (C->stub_function() == NULL) {
+  //   st->print("nmethod entry barrier\n\t");
+  // }
 }
 #endif
 
 void MachPrologNode::emit(CodeBuffer &cbuf, PhaseRegAlloc *ra_) const {
   Compile* C = ra_->C;
   C2_MacroAssembler _masm(&cbuf);
+  // Register nmethod_tmp = Z_R3;
```

Comment:
Don't kill R3! I'd use R0 and R1 only.

Reply:
I would guess this is because, even though it's a volatile register, R3 can be a parameter or return value. So in the context of the compiler, we really don't want to use this register?

BTW, I have now removed my changes to this file.

Reply:
R3 can contain a parameter at this point. You don't want to overwrite it.

Reply:
Makes sense. Thanks.